dns.c

```diff
@@ -20,13 +20,17 @@ extern int res_search();
 #include "stralloc.h"
 #include "case.h"
 
+#define MAX_EDNS_RESPONSE_SIZE 65536
+
 static unsigned short getshort(c) unsigned char *c;
 { unsigned short u; u = c[0]; return (u << 8) + c[1]; }
 
-static union { HEADER hdr; unsigned char buf[PACKETSZ]; } response;
+static struct { unsigned char *buf; } response;
```

Comment:
Indeed, union does not support [flexible array member](https://en.wikipedia.org/wiki/Flexible_array_member).

What about degenerating it entirely to `char *` or even `void *` or even plain `HEADER`?

Reply:
second thought, with a `static HEADER *responsebuf;`, responsebuf = malloc(PACKETSZ+1);` would not need cast anywhere (even in `lookup`, but just because it is not fully prototyped).

Reply:
Might be good. Happy to see it get tried :-)